src/js/core.js

```diff
@@ -14,16 +14,49 @@
     }
 
     function handleDisabledEnterKeydown(event, element) {
-        if (this.options.disableReturn || element.getAttribute('data-disable-return')) {
+        if (!this.options.singleEnterBlockElement) {
+            var p = MediumEditor.selection.getSelectionStart(this.options.ownerDocument),
+                caretPositions = MediumEditor.selection.getCaretOffsets(p),
+                nodeHtml = p.innerHTML,
+                textLeft,
+                textRight,
+                newEl;
+
+            if (p.nodeName.toLowerCase() !== 'p') {
+                return;
+            }
+
+            textLeft = nodeHtml.substring(0, caretPositions.left);
+            textRight = nodeHtml.substring(caretPositions.left, nodeHtml.length);
+
             event.preventDefault();
-        } else if (this.options.disableDoubleReturn || element.getAttribute('data-disable-double-return')) {
-            var node = MediumEditor.selection.getSelectionStart(this.options.ownerDocument);
 
-            // if current text selection is empty OR previous sibling text is empty OR it is not a list
-            if ((node && node.textContent.trim() === '' && node.nodeName.toLowerCase() !== 'li') ||
-                (node.previousElementSibling && node.previousElementSibling.nodeName.toLowerCase() !== 'br' &&
-                 node.previousElementSibling.textContent.trim() === '')) {
+            //if enter is being pressed twice
+            if (textRight.substring(0, 4) === '<br>') {
+                //need to remove the last added br tag, and add a new p tag
+                p.innerHTML = textLeft;
+                var newPElement = document.createElement('p');
+                textRight = textRight.substring(4, textRight.length);
+                newPElement.innerHTML = textRight;
+                p.parentNode.insertBefore(newPElement, p.nextSibling);
```

Comment:
Similar to feedback for #879, it would be great if we could accomplish this "wrap the text in a new `<p>`" via something like `MediumEditor.util.execFormatBlock()` so that we lean on the browser to handle it correctly, and we do it in a way that will work for undo/redo.